File: training/go/domain/model/creature_type.go
package model

type CreatureType int

const (
	Monstrosity CreatureType = iota
	Beast
	Aberration
	Celestial
	Dragon
	Fiend
	Humanoid
	Undead
)

func (t CreatureType) String() string {
	return [...]string{"monstrosity", "beast", "aberration", "celestial", "dragon", "fiend", "humanoid", "undead"}[t]
}

File: sample-projects/ruby/lib/de/sots/cellars_and_centaurs/domain/model/creature_type.rb
# frozen_string_literal: true

require_relative '../../version'

module De::Sots::CellarsAndCentaurs::Domain::Model
  module CreatureType
    MONSTROSITY = :monstrosity
    BEAST = :beast
    ABERRATION = :aberration
    CELESTIAL = :celestial
    DRAGON = :dragon
    FIEND = :fiend
    HUMANOID = :humanoid
    UNDEAD = :undead

    ALL = [MONSTROSITY, BEAST, ABERRATION, CELESTIAL, DRAGON, FIEND, HUMANOID, UNDEAD].freeze

    def self.valid?(creature_type)
      ALL.include?(creature_type)
    end
  end
end
